server/src/main/java/org/opensearch/index/codec/composite/Composite99DocValuesWriter.java

```diff
@@ -95,23 +112,91 @@ private void createCompositeIndicesIfPossible(DocValuesProducer valuesProducer,
             fieldProducerMap.put(field.name, valuesProducer);
             compositeFieldSet.remove(field.name);
         }
+        segmentFieldSet.remove(field.name);
+        if (segmentFieldSet.isEmpty()) {
```

Comment:
We still have 2 modes here, where star tree is not created, and one where it is created with null values.
Lets check for interesection of SortedNumericDV fields, and StarTree fields, and short-circuit in cases where the interesection set is empty

Reply:
ack , will pick it up in this if there are any more comments or in off heap PR since gradle check is passing.

Reply:
Addressing this in off heap PR.